pkg/apis/numaplane/v1alpha1/status_types.go

```diff
@@ -27,19 +27,26 @@ import (
 // ConditionType is a valid value of Condition.Type
 type ConditionType string
 
-// +kubebuilder:validation:Enum="";Pending;Running;Failed;NotApplicable
+// +kubebuilder:validation:Enum="";Pending;Deployed;Failed
 type Phase string
 
 const (
+	// PhasePending indicates that a reconciliation operation on the rollout spec has started.
+	// In this phase, the reconciliation process could take some time and/or happen with multiple reconciliation calls.
 	PhasePending Phase = "Pending"
-	PhaseRunning Phase = "Running"
-	PhaseFailed  Phase = "Failed"
-	PhaseNA      Phase = "NotApplicable"
 
-	// ConditionConfigured indicates valid configuration.
-	ConditionConfigured ConditionType = "Configured"
+	// PhaseDeployed indicates that the child resource has been applied to the cluster.
+	PhaseDeployed Phase = "Deployed"
 
-	ConditionChildResourcesHealthy ConditionType = "ChildResourcesHealthy"
+	// PhaseFailed indicates that one or more errors have occurred during reconciliation.
+	PhaseFailed Phase = "Failed"
+
+	// ConditionChildResourceHealthy indicates if the child resource is in a healthy phase (ex: not pending, not paused, not deleting, etc.).
+	// Child health is set to True only if the child resource generation is equal to its own observedGeneration.
+	ConditionChildResourceHealthy ConditionType = "ChildResourcesHealthy"
+
+	// ConditionChildResourceDeployed indicates that the child resource was deployed.
+	ConditionChildResourceDeployed ConditionType = "ChildResourceDeployed"
```

Comment:
We are going to add pausing/paused condition to pipelinerollout, right?

Reply:
At this time, if the pipeline is pausing or paused I set the child resource condition as unhealthy [here](https://github.com/numaproj/numaplane/blob/observedGeneration/internal/controller/pipelinerollout_controller.go#L283). If we want to show in ArgoCD the health as `suspended` similar to [jobs](https://argo-cd.readthedocs.io/en/stable/operator-manual/health/#job) using custom LUA script, then I can create a condition for the pipeline status to say a pipeline is paused or pausing and that can be used in the LUA script.

Reply:
Introduced pause or pausing condition.